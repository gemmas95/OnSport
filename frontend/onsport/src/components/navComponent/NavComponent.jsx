import React from 'react';
import { NavLink } from 'react-router-dom';
import './navComponent.scss';
import LoginButton from '../auth/loginComponent/LoginButton';
import LogoutButton from '../auth/logoutButton/LogoutButton';
import { useAuth0 } from '@auth0/auth0-react';

function NavComponent() {
    const { user, isAuthenticated } = useAuth0();

    return (
        <header className="nav__alignment">
            <h1 className="logoPage">OnSport</h1>
            <NavLink to="/">Sports</NavLink>
            {!isAuthenticated && <LoginButton />}
            {isAuthenticated && (
                <>
                    <NavLink to="/auth/profile">
                        <img
                            className="profileImg"
                            src={
                                (user && user.picture) ||
                                'https://image.flaticon.com/icons/svg/848/848043.svg'
                            }
                            alt="Profile"
                        ></img>
                    </NavLink>
                    <LogoutButton />
                </>
            )}
        </header>
    );
}

export default NavComponent;
