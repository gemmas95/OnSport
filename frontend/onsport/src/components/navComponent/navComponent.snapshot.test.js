import renderer from 'react-test-renderer';
import React from 'react';
import { BrowserRouter as Router } from 'react-router-dom';
import NavComponent from './NavComponent';

describe('NavComponent', () => {
    const navCompoentTree = renderer.create(
        <Router>
            <NavComponent />
        </Router>
    );
    it('should match', () => {
        expect(navCompoentTree).toMatchSnapshot();
    });
});
