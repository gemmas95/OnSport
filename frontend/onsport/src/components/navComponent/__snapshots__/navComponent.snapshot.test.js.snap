// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NavComponent should match 1`] = `
<header
  className="nav__alignment"
>
  <h1
    className="logoPage"
  >
    OnSport
  </h1>
  <a
    aria-current="page"
    className="active"
    href="/"
    onClick={[Function]}
    style={Object {}}
  >
    Sports
  </a>
  <button
    onClick={[Function]}
  >
    Log In
  </button>
</header>
`;
